mythril/analysis/modules/integer.py

```diff
@@ -88,66 +90,83 @@ def execute(self, state: GlobalState):
-            self._handle_sub(state)
-        elif state.get_current_instruction()["opcode"] == "SSTORE":
-            self._handle_sstore(state)
-        elif state.get_current_instruction()["opcode"] == "JUMPI":
-            self._handle_jumpi(state)
-        elif state.get_current_instruction()["opcode"] in ("RETURN", "STOP"):
-            self._handle_transaction_end(state)
-
-    def _handle_add(self, state):
+        opcode = state.get_current_instruction()["opcode"]
+        func = {
+            "ADD": self._handle_add,
+            "SUB": self._handle_sub,
+            "MUL": self._handle_mul,
+            "SSTORE": self._handle_sstore,
+            "JUMPI": self._handle_jumpi,
+            "RETURN": self._handle_transaction_end,
+            "STOP": self._handle_transaction_end,
+            "EXP": self._handle_exp,
+        }
+        func[opcode](state)
+
+    def _get_args(self, state):
         stack = state.mstate.stack
         op0, op1 = (
             self._make_bitvec_if_not(stack, -1),
             self._make_bitvec_if_not(stack, -2),
         )
+        return op0, op1
+
+    def _handle_add(self, state):
+        op0, op1 = self._get_args(state)
         c = Not(BVAddNoOverflow(op0, op1, False))
 
         # Check satisfiable
         model = self._try_constraints(state.node.constraints, [c])
         if model is None:
             return
 
-        annotation = OverUnderflowAnnotation(state, "addition", c)
+        annotation = OverUnderflowAnnotation(state, "addition", [c])
         op0.annotate(annotation)
 
     def _handle_mul(self, state):
-        stack = state.mstate.stack
-        op0, op1 = (
-            self._make_bitvec_if_not(stack, -1),
-            self._make_bitvec_if_not(stack, -2),
-        )
-
+        op0, op1 = self._get_args(state)
         c = Not(BVMulNoOverflow(op0, op1, False))
 
         # Check satisfiable
         model = self._try_constraints(state.node.constraints, [c])
         if model is None:
             return
 
-        annotation = OverUnderflowAnnotation(state, "multiplication", c)
+        annotation = OverUnderflowAnnotation(state, "multiplication", [c])
         op0.annotate(annotation)
 
     def _handle_sub(self, state):
-        stack = state.mstate.stack
-        op0, op1 = (
-            self._make_bitvec_if_not(stack, -1),
-            self._make_bitvec_if_not(stack, -2),
-        )
+        op0, op1 = self._get_args(state)
         c = Not(BVSubNoUnderflow(op0, op1, False))
 
         # Check satisfiable
         model = self._try_constraints(state.node.constraints, [c])
         if model is None:
             return
 
-        annotation = OverUnderflowAnnotation(state, "subtraction", c)
+        annotation = OverUnderflowAnnotation(state, "subtraction", [c])
+        op0.annotate(annotation)
+
+    def _handle_exp(self, state):
+        op0, op1 = self._get_args(state)
+        if op0.symbolic and op1.symbolic:
+            c1, c2 = (
```

Comment:
```suggestion
            constraint = And(
```
If we do that, then we don't need all the other changes which require a list of constraints vs just the one constraint